template/tasks.py

```diff
@@ -24,13 +24,17 @@ def help(ctx):
 
 
 @invoke.task
-def format(ctx):
+def format(ctx, check:bool = False) -> None:
     """
-    Apply automatic code formating tools
+    Apply automatic code formatting tools
+
+    By default, this modifies files to match coding style guidelines.
+    When `check` is True, it performs a dry-run to identify non-compliant
+    files without applying changes.
     """
     _title("Applying code formatters ")
-    ctx.run("poetry run black src")
-    ctx.run("poetry run isort src")
+    ctx.run(f"poetry run black src{' --check' if check else ''}")
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**thought (non-blocking):** I generally dislike to embed logic _within_ f-strings

This is sufficiently simple I'm not worried about it, but in general I like to limit f-string to variable insertion only rather than logic. See the `test` task for how we deal with building out optional arguments there.‏

Reply:
Thank you for your feedback. I’ve considered your suggestion and prefer to keep the current approach for its readability and straightforwardness. However, I’m open to revisiting this if it becomes a concern in the future.